src/command/build.rs

```diff
@@ -1,113 +1,92 @@
-//! Initializing a crate for packing `.wasm`s.
-
 use bindgen;
 use build;
-use command::utils::set_crate_path;
+use command::utils::{create_pkg_dir, set_crate_path};
 use emoji;
 use error::Error;
 use indicatif::HumanDuration;
 use manifest;
 use progressbar::Step;
 use readme;
 use slog::Logger;
-use std::fs;
-use std::path::{Path, PathBuf};
+use std::path::PathBuf;
 use std::time::Instant;
 use PBAR;
 
-/// Construct our `pkg` directory in the crate.
-pub fn create_pkg_dir(path: &Path, step: &Step) -> Result<(), Error> {
-    let msg = format!("{}Creating a pkg directory...", emoji::FOLDER);
-    PBAR.step(step, &msg);
-    let pkg_dir_path = path.join("pkg");
-    fs::create_dir_all(pkg_dir_path)?;
-    Ok(())
+/// Everything required to configure and run the `wasm-pack init` command.
+pub(crate) struct Build {
+    pub crate_path: PathBuf,
+    pub scope: Option<String>,
+    pub disable_dts: bool,
+    pub target: String,
+    pub debug: bool,
+    // build_config: Option<BuildConfig>,
+    pub crate_name: String,
 }
 
-/// The `InitMode` determines which mode of initialization we are running, and
+/// The `BuildMode` determines which mode of initialization we are running, and
 /// what build and install steps we perform.
-pub enum InitMode {
+pub enum BuildMode {
     /// Perform all the build and install steps.
     Normal,
-    /// Don't build the crate as a `.wasm` but do install tools and create
-    /// meta-data.
-    Nobuild,
     /// Don't install tools like `wasm-bindgen`, just use the global
     /// environment's existing versions to do builds.
     Noinstall,
 }
 
-/// Everything required to configure and run the `wasm-pack init` command.
-pub struct Init {
-    crate_path: PathBuf,
-    scope: Option<String>,
-    disable_dts: bool,
-    target: String,
+/// Everything required to configure and run the `wasm-pack build` command.
+#[derive(Debug, StructOpt)]
+pub struct BuildOptions {
+    /// The path to the Rust crate.
+    #[structopt(parse(from_os_str))]
+    pub path: Option<PathBuf>,
+
+    /// The npm scope to use in package.json, if any.
+    #[structopt(long = "scope", short = "s")]
+    pub scope: Option<String>,
+
+    #[structopt(long = "mode", short = "m", default_value = "normal")]
+    /// Sets steps to be run. [possible values: no-install, normal]
+    pub mode: String,
+
+    #[structopt(long = "no-typescript")]
+    /// By default a *.d.ts file is generated for the generated JS file, but
+    /// this flag will disable generating this TypeScript file.
+    pub disable_dts: bool,
+
+    #[structopt(long = "target", short = "t", default_value = "browser")]
```

Comment:
btw `browser` is not accurate. ESM will be soon available in Nodejs, which isn't a browser.

The correct targets are:
- CJS (commonjs)
- ESM (ECMAScript modules)

Reply:
> ESM will be soon available in Nodejs, which isn't a browser.

so should I leave a FIXME note here or change it to `ESM` right now?

Reply:
leave whatever the current behavior is, and let's file an issue to update.